src/collections/collection.ts

```diff
@@ -346,18 +278,15 @@ export class Collection {
 
     async findOneAndUpdate(filter: Record<string, any>, update: Record<string, any>, options?: FindOneAndUpdateOptions): Promise<JSONAPIModifyResult> {
         return executeOperation(async (): Promise<JSONAPIModifyResult> => {
-            const command: FindOneAndUpdateCommand = {
+            const command = {
                 findOneAndUpdate: {
                     filter,
                     update,
-                    options
+                    options,
+                    ...(options?.sort != null ? { sort: options?.sort } : {})
                 }
             };
             setDefaultIdForUpsert(command.findOneAndUpdate);
-            if (options?.sort) {
-                command.findOneAndUpdate.sort = options.sort;
-                delete options.sort;
```

Comment:
And here as well? (i.e. Are we missing this `delete` ?)

Reply:
Fixed in 70a98a1